File: backend/src/main/java/de/neuefische/backend/reward/Reward.java
package de.neuefische.backend.reward;

import org.springframework.data.annotation.Id;

import java.time.LocalDateTime;

public record Reward(
        @Id
        String id,
        String name,
        String description,
        double price,
        double savingAllocated,
        boolean statusOpen,
        LocalDateTime rewardCreated
) {
}

File: frontend/src/Reward.ts
export type Reward = {
    id: string,
    name: string,
    description: string,
    price: number,
    savingAllocated: number,
    statusOpen: boolean,
    rewardCreated: Date
}
export type NewReward = {
    name: string,
    description: string,
    price: number,
}